src/printer.rs

```diff
@@ -753,78 +753,172 @@ impl Printer for InteractivePrinter<'_> {
-                                                    .iter()
-                                                    .map(|d| d
-                                                        .generate(line_number, true, self)
-                                                        .text)
-                                                    .collect::<Vec<String>>()
-                                                    .join(" ")
-                                            ))
-                                        } else {
-                                            Some("".to_string())
+                            match self.config.wrapping_mode {
+                                WrappingMode::Word => {
+                                    // Word wrapping: split text into words and wrap at word boundaries
+                                    let words: Vec<&str> = text.split_whitespace().collect();
+                                    let mut current_line = String::new();
+                                    let mut current_line_width = 0;
+
+                                    for word in words.iter() {
+                                        let word_width = word
+                                            .chars()
+                                            .map(|c| c.width().unwrap_or(0))
+                                            .sum::<usize>();
+
+                                        // Add space before word if it's not the first word in the line
+                                        let space_width =
+                                            if current_line.is_empty() { 0 } else { 1 };
+                                        let total_width = word_width + space_width;
+
+                                        // Check if the word fits on the current line
+                                        if current_line_width + total_width > max_width
+                                            && !current_line.is_empty()
+                                        {
+                                            // Generate wrap padding if not already generated.
+                                            if panel_wrap.is_none() {
+                                                panel_wrap = if self.panel_width > 0 {
+                                                    Some(format!(
+                                                        "{} ",
+                                                        self.decorations
+                                                            .iter()
+                                                            .map(|d| d
+                                                                .generate(line_number, true, self)
+                                                                .text)
+                                                            .collect::<Vec<String>>()
+                                                            .join(" ")
```

Comment:
I wonder, as the code above is doing a `split_whitespace()`, and here it is doing a join on space explicitly, will it lose formatting like tabs, non-breaking spaces, double spaces etc.? It would be nice to see more tests here.
Would it be quicker and easier to take a different approach, instead of splitting each word, to find the first whitespace before the terminal width and split there instead?
Also, I would personally be happier if this can be split out into smaller functions, because there is a lot of code here.